api/api/src/main/scala/uk/ac/wellcome/platform/api/swagger/SwaggerScalaModelConverter.scala

```diff
@@ -0,0 +1,204 @@
+/* IMPORTANT:
```

Comment:
Thought on the test in general.

Do you think with things like paths / query options etc we could get those from the routes, and make sure they're in the `SwaggerDocs.scala`?

What I am trying to guard against is that we add something new to the API and don't document it. The use of tis API is meant to be gearing towards being used by loads of people, so it's a consideration.

Reply:
:thinking: We would have to inspect the routes and have some rules about what routes need swagger documentation and those which don't (health checks, context.json, swagger.json itself etc).

We don't do anything like this for the pre-refactor Finatra version, so reckon this is beyond the scope of this PR.

Reply:
Yep, agree that is beyond the scope of this PR. 
Also, I'm not sure in practice if the hassle is actually worth it: we currently have effectively only one route (works) and although we plan to add items and others later on, we don't really know how that's going to look. It might even be that we split the api into multiple services and this becomes a non-issue. (Not saying we will definitely be doing it, just saying maybe inspecting the routes now is a bit over-engineering)

Reply:
Re-reading @jamesgorrie's comment, it seems he meant checking that all query parameters are documented rather than all routes. That is a bit easier, so have added a commit [here](https://github.com/wellcometrust/catalogue/pull/192/commits/780526f3ab89bb078c132283c7a34a84ffd68210).

Turned out to be a good idea, as it picked up a few we hadn't documented yet!

Reply:
Nice! I think this PR is massive though and it already adds plenty of value as is... we should get it merged and iterate in subsequent PRs, me thinks.

Reply:
https://github.com/wellcometrust/platform/issues/3967